runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -136,3 +136,30 @@ J9::ARM64::TreeEvaluator::checkcastEvaluator(TR::Node *node, TR::CodeGenerator *
 
    return targetRegister;
    }
+
+TR::Register *
+J9::ARM64::TreeEvaluator::arraylengthEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(cg->fe());
+   // ldrw R, [B + contiguousSize]        
+   // cmp R, 0                            
+   // cselw R, [B + discontiguousSize]    
+   //
+   TR::Register *objectReg = cg->evaluate(node->getFirstChild());
+   TR::Register *lengthReg = cg->allocateRegister();
+   TR::Register *discontiguousLengthReg = cg->allocateRegister();
+
+   TR::MemoryReference *contiguousArraySizeMR = new (cg->trHeapMemory()) TR::MemoryReference(objectReg, fej9->getOffsetOfContiguousArraySizeField(), 4, cg);
+   TR::MemoryReference *discontiguousArraySizeMR = new (cg->trHeapMemory()) TR::MemoryReference(objectReg, fej9->getOffsetOfDiscontiguousArraySizeField(), 4, cg);   
+
+   generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, lengthReg, contiguousArraySizeMR);
+   generateCompareImmInstruction(cg, node, lengthReg, 0);
+   generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, discontiguousLengthReg, discontiguousArraySizeMR);
+   generateCondTrg1Src2Instruction(cg, TR::InstOpCode::cselw, node, lengthReg, lengthReg, discontiguousLengthReg, TR::CC_EQ);
```

Comment:
I think the condition is backwards.  You want `TR::CC_NE` if you're using the registers in the `CSEL` in that order.  If the arraylength is zero, its true length is in the discontiguous length field.